cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/ExpressionHandler.kt

```diff
@@ -244,171 +242,31 @@ class ExpressionHandler(lang: JavaLanguageFrontend) :
-                if (typeString != null) {
-                    baseType = this.objectType(typeString)
-                } else {
-                    val name: String
-                    val tokenRange = scope.asNameExpr().tokenRange
-                    name =
-                        if (tokenRange.isPresent) {
-                            tokenRange.get().toString()
-                        } else {
-                            scope.asNameExpr().nameAsString
-                        }
-                    val qualifiedNameFromImports = frontend.getQualifiedNameFromImports(name)
-                    baseType =
-                        if (qualifiedNameFromImports != null) {
-                            this.objectType(qualifiedNameFromImports)
-                        } else {
-                            log.info("Unknown base type 1 for {}", fieldAccessExpr)
-                            unknownType()
-                        }
-                }
-            }
-            base = newReference(scope.asNameExpr().nameAsString, baseType, rawNode = scope)
-            base.isStaticAccess = isStaticAccess
-        } else if (scope.isFieldAccessExpr) {
-            base =
-                handle(scope) as de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression?
-                    ?: newProblemExpression("Could not parse base")
-            var tester = base
-            while (tester is MemberExpression) {
-                // we need to check if any base is only a static access, otherwise, this is a member
-                // access
-                // to this base
-                tester = tester.base
-            }
-            if (tester is Reference && tester.isStaticAccess) {
-                // try to get the name
-                val name: String
-                val tokenRange = scope.asFieldAccessExpr().tokenRange
-                name =
-                    if (tokenRange.isPresent) {
-                        tokenRange.get().toString()
-                    } else {
-                        scope.asFieldAccessExpr().nameAsString
-                    }
-                val qualifiedNameFromImports = frontend.getQualifiedNameFromImports(name)
-                val baseType =
-                    if (qualifiedNameFromImports != null) {
-                        this.objectType(qualifiedNameFromImports)
-                    } else {
-                        log.info("Unknown base type 2 for {}", fieldAccessExpr)
-                        unknownType()
-                    }
-                base =
-                    newReference(scope.asFieldAccessExpr().nameAsString, baseType, rawNode = scope)
-                base.isStaticAccess = true
-            }
-        } else {
-            base =
-                handle(scope) as de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression?
-                    ?: newProblemExpression("Could not parse base")
-        }
-        var fieldType: Type?
+
```

Comment:
Please document the method according to the new standards: https://github.com/Fraunhofer-AISEC/cpg/pull/1683#discussion_r1752563555